### DISCUSSION TITLE
Ankiとの連携

### DISCUSSION THREAD

**Discussion by Matt20240925** [wtetsu/mouse-dictionary#95] [General]
初めまして、米国に一時的に仕事で駐在しており、日々最適な英語学習を研究しています。
プログラミングコードの意味が1つも分からない状態から、公開なさっているMouse Dictionaryのコードを分析し、既存のMouse Dictionaryに「特定のキーを押したら、ポップアップ上に表示されている英単語とその意味をAnkiに飛ばし、新たなカードを作成する機能」を実装しました。

そこで一点相談ですが、機能の更なる充実のためにご助言いただけないでしょうか。
また、貴殿の承諾が下りるのであれば、Mouse Dictionaryをパワーアップさせた拡張機能をウェブストアで公開したく思っています。こちらについても意見いただければ幸いです。

**Comment by wtetsu** [comment]
私自身の意見としては、特殊な機能やなんらかのサービスに特化した機能は、Mouse Dictionaryそのものに手をいれるというよりは外側から追加するような方法を推奨します。

Ankiに関して言えば、[このようなもの](https://greasyfork.org/en/scripts/429029-mouse-dictionary-helper)をつくっている方がいるようです。これはMouse Dictionaryそのものを変更せずに実現しています。
(私はAnkiユーザではないので実際に試したことはありませんが)

Mouse Dictionary側としては、なんらかのサービスに特化した機能をひとつひとつ実装することは積極的でありませんが(キリがないので)、「ユーザが追加機能を実装しやすくなるようなる機能」を追加することには異存はありません。

----

> Mouse Dictionaryをパワーアップさせた拡張機能をウェブストアで公開したく

[ライセンスの範囲](https://github.com/wtetsu/mouse-dictionary/blob/master/LICENSE)で実施していただいて構いませんが、ダウンロードしようとする人が本家と混乱しないような名前にしていただけると助かります(たとえば「Mouse Dictionary改」のような名前だとユーザが混乱する可能性があるため)

また、私自身の意見をいうと「パワーアップした本体を公開」というよりは「ユーザがパワーアップ部分のみをつくれること」のほうが理想的だと考えています。

**Reply by Matt20240925** [reply]
早速のご返信誠にありがとうございます。また返信が遅くなり申し訳ありません。
外側から追加する方法については紹介いただいたMouse Dictionary Helperの事例によりイメージとして理解することができました。素人ながら私が追加で実装した機能は設定したショートカットキーを押すと、ホバリングしている語彙を取得し単語と意味をAnkiに送るものですが、同機能を外側から追加できるかという点は今後検討させていただきます。
ウェブストアでの公開につきましても非常によく理解いたしました。ありがとうございます。
